### PR TITLE
Fix VTMR: resolve conflicts with upstream

### PR DESCRIPTION
**Pull request by guokeno0**
@alainjobart 

This is the same PR which removes the fix for streaming query RowsAffected

### CODE REVIEW

**Comment by guokeno0** [comment]
@erzel

**Comment by guokeno0** [comment]
Need someone to look at this, the problem will fail internal MR jobs on all single-sharded keyspaces

**Comment by alainjobart** [comment]
Can we add a test somewhere for this? maybe tabletmanager.py would be a good place, create a new unit test there, you can setup the two tablets as master+rdonly in a single keyspace, and run vtgate.split_query on it (see custom_sharding.py for an example of how to do that). Thanks!

**Comment by guokeno0** [comment]
As discussed, I've changed SplitQuery to make decision by ShardingColumnType. I've also changed sandbox test to add ShardingColumnType to the fake topo.

**Comment by alainjobart** [comment]
LGTM I'll work on some tests for this when submitted.

File: go/vt/vtgate/vtgate.go [hunk 1]
```diff
@@ -605,7 +605,10 @@ func (vtg *VTGate) SplitQuery(ctx context.Context, keyspace string, sql string,
 		return nil, err
 	}
 	perShardSplitCount := int64(math.Ceil(float64(splitCount) / float64(len(shards))))
-	if srvKeyspace.ShardingColumnName != "" {
+	// sharding_column_type != KeyspaceIdType_UNSET can happen in one of the following two cases:
+	// 1. We are querying a sharded keyspace;
+	// 2. We are querying an unsharded keyspace which is being sharded.
```
Comment (erzel) [inline go/vt/vtgate/vtgate.go]: Thanks for the comment. It's clearer now.

**Pull request merged**